src/pages/api/comments.ts

```diff
@@ -0,0 +1,151 @@
+import { getSession } from "@auth0/nextjs-auth0";
+import { Prisma } from "@prisma/client";
+import { withSentry } from "@sentry/nextjs";
+import { NextApiHandler } from "next";
+
+import { prisma } from "../../shared/prisma-helper";
+import { generateAvatarUrl } from "../../shared/user-helpers";
+import {
+  CommentsApiHandlerSuccessfulGetResponseBody,
+  CommentsApiHandlerSuccessfulPostResponseBody,
+  NewComment,
+  PublicCommentInfo,
+} from "../../types";
+
+const commentSelect = {
+  accidentId: true,
+  author: {
+    select: {
+      name: true,
+      email: true,
+      picture: true,
+    },
+  },
+  createdAt: true,
+  id: true,
+  isPublished: true,
+  text: true,
+} as const;
+
+const convertSelectedCommentToPublicInfo = ({
+  accidentId,
+  author,
+  createdAt,
+  id,
+  isPublished,
+  text,
+}: Prisma.CommentGetPayload<{
+  select: typeof commentSelect;
+}>): PublicCommentInfo => ({
+  accidentId,
+  authorAvatarUrl: generateAvatarUrl(author.picture ?? undefined, author.email),
+  authorName: author.name,
+  createdAt: createdAt.toISOString(),
+  id,
+  isPublished,
+  text,
+});
+
+const handler: NextApiHandler = async (req, res) => {
```

Comment:
This handler was moved from `/api/comments/[accident-id]`. It is `POST /api/comments` or `GET /api/comments?accidentid=N`. We don’t have to be RESTful as out API is not meant to be public at this point, so I can split the route if the file looks too long.